apis/secretsmanager/v1alpha1/secret_types.go

```diff
@@ -0,0 +1,223 @@
+/*
+Copyright 2020 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/docker/swarmkit/api/timestamp"
+)
+
+// Keys used in connection secret.
+const (
+	ConnectionSecretKeyName        = "secret"
+	ConnectionSecretKeyProjectName = "projectName"
+)
+
+// SecretVersionState gives the state of a [SecretVersion][google.cloud.secretmanager.v1.SecretVersion], indicating if it can be accessed.
+type SecretVersionState int32
```

Comment:
I don't think we need to define all of the iterations of `SecretVersionState` as `const` since they are output only fields, we can just set it to whatever GCP returns to us

Reply:
Looks like we might need them for `SecretVersion` CRD though 👍